test/classes/ferguson/delete-free/rc4.chpl

```diff
@@ -0,0 +1,23 @@
+use SharedObject;
+
+class Impl {
+  var field: int;
+  proc init(arg:int) {
+    field = arg;
+    writeln("in Impl.init ", field);
+  }
+
+  proc ~Impl() {
+    writeln("in Impl.~Impl ", field);
+  }
+}
+
+var globalSharedObject:Shared(Impl);
+
+proc makeGlobalSharedObject() {
+  var x = new Shared(new Impl(1));
+
+  globalSharedObject = x;
+}
+
+makeGlobalSharedObject();
```

Comment:
let's add writeln(globalSharedObject.p.field)
this will for example show that deinit happens later